extractor/text.go

```diff
@@ -1180,17 +1181,46 @@ var (
 // sortPosition sorts a text list by its elements' positions on a page.
 // Sorting is by orientation then top to bottom, left to right when page is orientated so that text
 // is horizontal.
+// Text is considered to be on different lines if the lines' orientedStart.Y differs by more than `tol`.
 func (pt *PageText) sortPosition(tol float64) {
+	if len(pt.marks) == 0 {
+		return
+	}
 	sort.SliceStable(pt.marks, func(i, j int) bool {
```

Comment:
Is this some kind of pre-sorting?
Followed by grouping into lines and finally sorting by yOrder (which is related to line number?).

Reply:
Yes. The pre-sorting is used to group the text into lines. yOrder is the line number.